src/promptflow/promptflow/_core/tracer.py

```diff
@@ -143,6 +147,37 @@ def _pop(self, output=None, error: Optional[Exception] = None):
         else:
             return output
 
+    @classmethod
+    def update_openai_tokens(cls, output):
```

Comment:
We are going to remove tracer in the future, maybe we can have another singleton class to collect and compute this.

Reply:
The operations may include the following:
```
SpanCollector.add_span(span)  # Add the span to the collector
SpanCollector.update_span_attribute(span_id, key, val)  # Update the token value in child span
SpanCollector.remove_span(span_id)  # Remove this span to avoid memory leak
```

Reply:
Refined